samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/ui/ConsentModalBottomSheetScreen.kt

```diff
@@ -0,0 +1,158 @@
+                dataElementsToRequest[de.attribute.identifier] = intentToRetain
+            }
+            namespacesToRequest[ns.namespace] = dataElementsToRequest
+        }
+        val encodedSessionTranscript = Cbor.encode(CborMap.builder().put("doesn't", "matter").end().build())
+        val encodedDeviceRequest = DeviceRequestGenerator(encodedSessionTranscript)
+            .addDocumentRequest(
+                request.mdocRequest!!.docType,
+                namespacesToRequest,
+                null,
+                null,
+                Algorithm.UNSET,
+                null
+            ).generate()
+        val deviceRequest = DeviceRequestParser(encodedDeviceRequest, encodedSessionTranscript)
+            .parse()
+
+        val docTypeRepo = DocumentTypeRepository()
+        docTypeRepo.addDocumentType(DrivingLicense.getDocumentType())
+        MdocConsentField.generateConsentFields(
+            deviceRequest.docRequests[0],
+            docTypeRepo,
+            null
+        )
+    }
+
+    var cardArt by remember {
+        mutableStateOf(ByteArray(0))
+    }
+    var relyingPartyDisplayIcon by remember {
+        mutableStateOf(ByteArray(0))
+    }
+    LaunchedEffect(Unit) {
+        cardArt = Res.readBytes("files/utopia_driving_license_card_art.png")
+        relyingPartyDisplayIcon = Res.readBytes("files/utopia-brewery.png")
+        sheetState.show()
+    }
+
+    val relyingParty = when (verifierType) {
+        VerifierType.KNOWN_VERIFIER -> {
+            ConsentRelyingParty(
+                trustPoint = TrustPoint(
+                    certificate = X509Cert.fromPem(IACA_CERT_PEM),
+                    displayName = "Utopia Brewery",
+                    displayIcon = relyingPartyDisplayIcon
+                ),
+                websiteOrigin = null,
+            )
+        }
+        VerifierType.UNKNOWN_VERIFIER_PROXIMITY ->  {
+            ConsentRelyingParty(
+                trustPoint = null,
+                websiteOrigin = null,
+            )
+        }
+        VerifierType.UNKNOWN_VERIFIER_WEBSITE ->  {
+            ConsentRelyingParty(
+                trustPoint = null,
+                websiteOrigin = "https://www.example.com",
+            )
+        }
+    }
+
+    if (sheetState.isVisible && cardArt.size > 0) {
+        ConsentModalBottomSheet(
+            sheetState = sheetState,
+            consentFields = consentFields,
+            ConsentDocument(
+                name = "Erika's Driving License",
+                cardArt = cardArt,
+                description = "Driving License",
+            ),
+            relyingParty = relyingParty,
+            onConfirm = {
+                scope.launch {
+                    sheetState.hide()
+                    onSheetConfirmed()
+                }
+            },
+            onCancel = {
+                scope.launch {
+                    sheetState.hide()
+                    showToast("The sheet was dismissed")
```

Comment:
Is this for debugging, or is this a message we intend to display to the user? Would they understand what we mean by "sheet"?

Reply:
It's just to give the user of the testapp an indication that the two buttons do something different.